### PR TITLE
feat(event/tag): add custom tag for resource and build trace link for…

### PR DESCRIPTION
**Pull request by xuqingyun**
### Description

- support to add custom tag for span, and get tag value from json path
- based on object's related resource tag, we can add redirect link to related object
  -  for example, if a pod span has a tag key nodes and tag value is node name, we can add link for tag nodes with linkPatterns configuration in [jaeger-ui](https://www.jaegertracing.io/docs/1.43/frontend-ui/)：
```
      linkPatterns: [
          "type": "tags",
          "key": "nodes",
          "url": "/redirect?cluster=#{cluster}&group=&name=#{nodes}&resource=nodes&ts=#{timeStamp}",
          "text": "#{nodes}"
        }
      ],
```  

image

### CODE REVIEW

File: pkg/aggregator/event/decorator/tag/resource_tag.go [hunk 1]
```diff
@@ -0,0 +1,188 @@
...
+}
+
+func (options *options) EnableFlag() *bool { return &options.enable }
+
+type tagDecorator struct {
+	options            options
+	logger             logrus.FieldLogger
+	objectCache        objectcache.ObjectCache
+	eventDecorator     decorator.UnionEventDecorator
+	ctx                context.Context
```
Comment (SOF3) [inline pkg/aggregator/event/decorator/tag/resource_tag.go:63]: field is unused

File: pkg/aggregator/event/event.go [hunk 2]
```diff
@@ -42,7 +42,7 @@ func NewEvent(field string, title string, when time.Time, traceSource string) *E
 	}
```
Comment (SOF3) [inline pkg/aggregator/event/event.go:42]: Rename `NewEvent` to `New`?

File: pkg/aggregator/event/event.go [hunk 3]
```diff
@@ -12,7 +12,7 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-package aggregator
+package event
```
Comment (SOF3) [inline pkg/aggregator/event/event.go:15]: What about naming this package `aggregatorevent` to reduce collision?

File: pkg/frontend/tf/config/default/default.go [hunk 4]
```diff
@@ -28,9 +30,23 @@ func init() {
 	manager.Global.ProvideMuxImpl("jaeger-transform-config/default", newProvider, tfconfig.Provider.DefaultId)
 }
 
+type options struct {
+	frontendDomain     string
+	resourceTagsToLink []string
+}
+
+func (options *options) Setup(fs *pflag.FlagSet) {
+	fs.StringVar(&options.frontendDomain, "jaeger-frontend-domain", "", "jaeger frontend domain e.g. http://localhost:3000")
```
Comment (SOF3) [inline pkg/frontend/tf/config/default/default.go:39]: It doesn't necessarily need to be a domain; Jaeger UI also supports non-root base path (through `--query.base-path`).
```suggestion
	fs.StringVar(&options.frontendDomain, "jaeger-frontend-base-url", "", "jaeger frontend URL prefix e.g. http://localhost:3000 or https://example.com/kelemetry, used for link generation")
```

File: pkg/frontend/tf/config/default/default.go [hunk 5]
```diff
@@ -28,9 +30,23 @@ func init() {
...
+	fs.StringSliceVar(&options.resourceTagsToLink, "jaeger-transform-resource-link", []string{"nodes"},
+		"add resource link for span with these tags. e.g. 'nodes, apiservices.apiregistration.k8s.io'")
```
Comment (SOF3) [inline pkg/frontend/tf/config/default/default.go:41]:
```suggestion
	fs.StringSliceVar(
		&options.resourceTagsToLink,
		"jaeger-transform-resource-link",
		[]string{"nodes"},
		"add resource link for span with these tags. e.g. 'nodes, apiservices.apiregistration.k8s.io'",
	)
```

File: pkg/frontend/tf/step/resource_link.go [hunk 6]
```diff
@@ -0,0 +1,88 @@
...
+				"resource": []string{gv.Resource},
+				"name":     []string{resourceKv.VStr},
+				"ts":       []string{timeStamp},
+			}.Encode()
+			span.Tags = append(span.Tags, model.KeyValue{
+				Key:  resourceKey + "_trace",
+				VStr: fmt.Sprintf("%v/redirect?%v", strings.TrimSuffix(visitor.Domain, "/"), query),
+			})
+		}
+	}
```
Comment (SOF3) [inline pkg/frontend/tf/step/resource_link.go:63]: Automatic hyperlinks are possible with a custom `linkPattern` in the Jaeger UI config, can we do anything about this?

**Review by SOF3 — COMMENTED** [review]
please merge `main`

**Comment by xuqingyun** [comment]
update

File: pkg/frontend/tf/step/resource_link.go (hunk 6 above)
Comment (xuqingyun) [reply pkg/frontend/tf/step/resource_link.go]: good idea

**Review by SOF3 — APPROVED** [review]

**Pull request merged**